rotkehlchen/chain/ethereum/airdrops.py

```diff
@@ -282,9 +289,36 @@ def get_poap_airdrop_data(name: str, data_dir: Path) -> dict[str, Any]:
     return data_dict
 
 
+def is_airdrop_claimed(
+        address: ChecksumEvmAddress,
+        amount: FVal,
+        asset: Asset,
+        database: DBHandler,
+) -> bool:
+    """Checks if an airdrop has already been claimed by checking if there
+    is an airdrop claim event for the given address, amount and asset."""
+    db_filter = EvmEventFilterQuery.make(
+        event_types=[HistoryEventType.RECEIVE],
+        event_subtypes=[HistoryEventSubType.AIRDROP],
+        addresses=[address],
+        assets=(asset,),
+    )
+    with database.conn.read_ctx() as cursor:
+        events = DBHistoryEvents(database).get_history_events(
+            cursor=cursor,
+            filter_query=db_filter,
+            has_premium=True,  # no need to limit filtered events here
+        )
+        for event in events:
+            if event.balance.amount == amount:
+                return True
```

Comment:
to avoid this loop you could add a filter on the amounts to the superclass of `EvmEventFilterQuery` and then use `get_history_events_count` so you could just check that 

```
return DBHistoryEvents(database).get_history_events_count(cursor, db_filter)[0] != 0
```